src/pages/index.js

```diff
@@ -1,140 +1,245 @@
+  const card = createCard(cardData);
+  cardSection.addItem(card);
 }
-
+//Create Card Function
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButton
+  );
+  return card.getCardElement();
+}
+//Click Card Function
 function handleImageClick(link, name) {
   popupWithImage.open({ name, link });
 }
 
-// Form Validators
-const profileEditValidator = new FormValidator(config, profileEditForm);
-const addCardValidator = new FormValidator(config, addCardFormElement);
-profileEditValidator.enableValidation();
-addCardValidator.enableValidation();
-
-// Event Handlers
-function fillEditProfileForm() {
-  profileTitleInput.value = profileTitle.textContent;
-  profileDescriptionInput.value = profileDescription.textContent;
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+      popupInstance.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
 }
 
-// Event Listeners
+function handleEditProfileSubmit(inputValues) {
+  function makeRequest() {
+    return api.updateUserInfo(inputValues).then((userData) => {
+      userInfo.setUserInfo(userData);
+    });
+  }
+  handleSubmit(makeRequest, profileEditModal);
+}
+
+function handleAddCardSubmit(inputValues) {
+  function makeRequest() {
+    return api.createCard(inputValues).then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+    });
+  }
+  handleSubmit(makeRequest, addCardPopup);
+}
+
+function handleDeleteButton(card) {
+  deleteImageModal.open();
+  return deleteImageModal.handleDelete(() => {
+    function makeRequest() {
+      return api.deleteCard(card.id).then(() => {
+        card.deleteCard();
+      });
+    }
+    handleSubmit(makeRequest, deleteImageModal);
+  });
+}
+
+function handleLikeButton(card) {
+  if (card.isLiked) {
+    api
+      .deleteLike(card.id)
+      .then(() => {
+        card.handleIsLiked(false);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card.id)
+      .then(() => {
+        card.handleIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+}
+
+function handleChangeProfilePicture(inputUrl) {
+  function makeRequest() {
+    return api.updateProfileImage(inputUrl).then((res) => {
+      userInfo.setProfileImage(res);
+    });
+  }
+  handleSubmit(makeRequest, profileImageModal);
+}
+
+//ProfileEditButtonEventListeners
 profileEditButton.addEventListener("click", () => {
-  profileEditValidator.hideInputError(profileEditForm);
-  fillEditProfileForm();
-  profileEditPopup.open();
+  formValidators["profile-form"].hideInputError(profileEditForm);
```

Comment:
`hideInputError` should not be called in `index.js` because everything is done inside `resetValidation`